### DISCUSSION TITLE
Plotly with filter checkboxes is trimmed off on left edge in HTML output

### DISCUSSION THREAD

**Discussion by Tb0nes** [quarto-dev/quarto-cli#6621] [Q&A] [answered]
### Description

Hello, 
Pardon my messy formatting. I'm trying to create a scatter plot with responsive data filter check boxes. When knitting to HTML, the left edge trims off. Neither of the following codes seem to work:

- rmarkdown chunk options: {r, fig.width=, fig.height=}
- quarto render options: #| fig-width: & #| fig-height:
- ggplotly: ggplotly(width=, height=)
- plotly layout: ggplotly() %>% layout(width=, height=)
- YAML: fig-responsive: false

Here's an example of the issue with the penguins data set for reproducibility:

````
---
format:
  html: 
    grid:
      sidebar-width: 300px
      body-width: 900px
      margin-width: 300px
      gutter-width: 1.5rem
fig-responsive: false
editor_options: 
  chunk_output_type: console
---


```{r, label=packages, results='hide', message=FALSE, echo=FALSE}

library(tidyverse)
library(plotly)
library(knitr)
library(crosstalk)
```


```{r, label=data, echo=FALSE}
library(palmerpenguins)
df <- penguins 
df <- drop_na(penguins)
```


INDIVIDUAL PLOT
```{r, label=individual_plot, echo=FALSE}

gg <- df %>% ggplot(aes(x = bill_length_mm, y = bill_depth_mm, color = species)) +
  geom_jitter()

gg 
```

PLOT W FILTER
```{r, label=plot_w_filter, echo=FALSE}
#Create new shared data for crosstalk library
shared_data <- df %>% 
  select(bill_depth_mm, bill_length_mm, species) %>%  
  SharedData$new()

#Create scatter plotly column
plty <- shared_data %>% 
  plot_ly(x = ~bill_length_mm, y = ~bill_depth_mm, color = ~species, type="scatter", mode="markers") 

#Combine filter(left) with scatter plotly (right)
flt_pltl <-bscols(widths = c(2,10), #<-(col1,col2) widths
       list(
            filter_checkbox(id = "species", label = "Species", sharedData = shared_data, group = ~species)),plty)

flt_pltl
```

````

Here are the knitter plot outputs:
gg
<img width="718" alt="plot_w_filter" src="https://github.com/quarto-dev/quarto-cli/assets/143238303/70352410-6e66-4583-b92d-7c664bb0c307">

flt_pltl
<img width="718" alt="plot_w_filter" src="https://github.com/quarto-dev/quarto-cli/assets/143238303/df2d9c7a-9527-4312-bc56-9020797082b2">

However, it appears perfectly fine when rendered in the RStudio console output viewer
<img width="505" alt="console" src="https://github.com/quarto-dev/quarto-cli/assets/143238303/25031ef9-44e5-480e-a3d6-370fdd407e01">

Any help would be greatly appreciated

**Comment by mcanouil** [comment]
You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by Tb0nes** [reply]
Now updated, thanks for the tip!

**Reply by mcanouil** [reply, marked as answer]
The issue comes from the bootstrap embedded by PlotLy
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/81608f5d-fb43-464b-a35e-8e414b369805">

See from the developper mode of a browser.
So you have to use the library to make sure it outputs something correct or rewrite the CSS.

```qmd
include-in-header:
  - text: |
      <style>
      .container-fluid.crosstalk-bscols {
        margin-left: 0;
        margin-right: 0;
      }
      </style>
```

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/1096a5dd-d155-4a69-91ab-9eab4e3051c8">

**Reply by Tb0nes** [reply]
@mcanouil This is excellent. It worked, thanks so much!
